runtime/vm/exceptionsupport.c

```diff
@@ -519,6 +519,16 @@ internalSetCurrentExceptionWithCause(J9VMThread *currentThread, UDATA exceptionN
 	currentThread->currentException = NULL;
 	currentThread->privateFlags &= ~J9_PRIVATE_FLAGS_REPORT_EXCEPTION_THROW;
 
+	if (J9VMCONSTANTPOOL_JAVALANGUNSUPPORTEDCLASSVERSIONERROR == exceptionNumber) {
+		J9JavaVM *vm = currentThread->javaVM;
+		if (J9_ARE_NO_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_CLASS_OBJECT_ASSIGNED)) {
+			/* This is JVM startup stage, exit with message now. */
+			PORT_ACCESS_FROM_JAVAVM(vm);
+			j9tty_err_printf(PORTLIB, "%s\n", utfMessage);
+			goto done;
```

Comment:
Is the intention this falls out to the normal unhandled exception path?  Or should a hard exit be considered?

Reply:
Yes, the intention is to fall out the normal unhandled exception path and avoid any class loading: https://github.com/eclipse/openj9/blob/5baeb841ecfdb62aab46df0fa436e1b0c2759256/runtime/vm/exceptionsupport.c#L648
A hard `exit(-1)` gives output like:
```
JVMCFRE003 bad major version; class=java/lang/Object, offset=6
```
and this PR provides
```
JVMCFRE003 bad major version; class=java/lang/Object, offset=6
JVMJ9VM019E Unrecoverable error: Unable to find and initialize required class java/lang/Object
JVMJ9VM023I This may indicate that JAVA_HOME is incorrect, or that class libraries are not installed
```
which I feel offer more details about such fatal error.